atc/db/job_factory.go

```diff
@@ -62,21 +62,14 @@ func (resources SchedulerResources) Lookup(name string) (SchedulerResource, bool
 }
 
 func (j *jobFactory) JobsToSchedule() (SchedulerJobs, error) {
-	tx, err := j.conn.Begin()
```

Comment:
Hi @clarafu, I think you are the DB expert. Here I removed the transaction because it cause the db query to hang at https://github.com/concourse/concourse/pull/6079/files#diff-61e6f1d527ec479d4312e73f52b3f93aR99 for unit test. But I want to understand why transaction is used here, this function does read-only, how a transaction helps?

Reply:
With the second commit, this is no longer an issue.